### ISSUE TITLE
i have a question , the data do not  refresh

### ISSUE THREAD

**Issue by leisure520** [Norbert515/flutter_list_drag_and_drop#3]
```
import 'package:flutter/material.dart';
import 'package:flutter_list_drag_and_drop/drag_and_drop_list.dart';

void main() {
  runApp(new TestApp());
}

class TestApp extends StatelessWidget {
  TestApp({Key key});

  // This widget is the root of your application.
  @override
  Widget build(BuildContext context) {
    return new MaterialApp(
      title: 'Flutter Demo',
      theme: new ThemeData(
          primarySwatch: Colors.blue, primaryColor: new Color(0xffd05c6b)),
      home: new MyHomePage(
        title: 'Flutter Demo Home Page',
        key: key,
      ),
    );
  }
}

class MyHomePage extends StatefulWidget {
  MyHomePage({Key key, this.title}) : super(key: key);
  final String title;

  @override
  MyHomePageState createState() => new MyHomePageState();
}

class MyHomePageState extends State<MyHomePage> {
  List<String> items = [
    '0',
    '1',
    '2',
    '3',
    '4',
    '5',
    '6',
    '7',
    '8',
    '9',
    '10',
    '11',
    '12',
    '13',
    '14',
    '15',
  ];
void add (){
    setState(() {
         items.add('value');     
        });

  }
  @override
  Widget build(BuildContext context) {
    return new Scaffold(
      appBar: new AppBar(
        title: new Text(widget.title),
      ),
      body: new DragAndDropList<String>(
        items,
        itemBuilder: (BuildContext context, item) {
          return new SizedBox(
            child: new Card(
              child: new ListTile(
                title: new Text(item),
              ),
            ),
          );
        },
        onDragFinish: (before, after) {
          String data = items[before];
          items.removeAt(before);
          items.insert(after, data);
        },
        canAccept: (one, two) => true,
        dragElevation: 18.0,
      ),
       floatingActionButton: new FloatingActionButton(
          onPressed: add,
          child: new Icon(Icons.add),
        )
    );
  }
}
```

**Comment by leisure520** [comment]
when i click the  button, the DragAndDropList  do not  add the string 'value'.

**Comment by Norbert515** [comment]
The drag_and_drop_list.dart needs this method:
```
  @override
  void didUpdateWidget(DragAndDropList<T> oldWidget) {
    super.didUpdateWidget(oldWidget);
    List data = widget.rowsData;
    rows = data.map((it) => new Data<T>(it)).toList();
    key.clear();
    for (int i = 0; i < rows.length; i++) {
      key.add(new GlobalKey());
    }
  }
```
I will update and refactor the library today and publish it on pub.

**Comment by leisure520** [comment]
thanks a lot,i try it.

**Comment by leisure520** [comment]
it works well ,thanks again

**Comment by Skquark** [comment]
I'm still having this problem when adding to the list, not sure if you published the change, but I didn't see the didUpdateWidget in mine, and when I tried to add it, I got this error:
Error: Getter not found: 'key'.
compiler message:     key.clear();
and then some related to that, so maybe I put it in the wrong spot, seemed it goes in your _DragAndDropListState, but didn't want to mess with it too much.  Maybe I'm getting that because I've got the use of the DragAndDropList where I assigned the key, but that that was needed for the Firestore fix.  It does dynamically add the new item in my firestore stream version, but not when using simple list.  Got things working well otherwise, thanks
